runtime/libpgmath/lib/common/arm64intrin.h

```diff
@@ -0,0 +1,2823 @@
+    unsigned char   c[16];
+  } __attribute__((aligned(16)));
+
+  s128iType type;
+
+  __s128i() : b(0), type(UnsignedInt) { }
+
+  __s128i(const float f[4]) : b(0), type(SignedInt) {
+    xird[0] = f[0];
+    xird[1] = f[1];
+    xird[2] = f[2];
+    xird[3] = f[3];
+  }
+
+  __s128i(const double d[2]) : b(0), type(SignedLong) {
+    xlrd[0] = d[0];
+    xlrd[1] = d[1];
+  }
+
+  __s128i(const int i[4]) : b(0), type(SignedInt) {
+    xird[0] = i[0];
+    xird[1] = i[1];
+    xird[2] = i[2];
+    xird[3] = i[3];
+  }
+
+  __s128i(const unsigned int i[4]) : b(0), type(UnsignedInt) {
+    xui[0] = i[0];
+    xui[1] = i[1];
+    xui[2] = i[2];
+    xui[3] = i[3];
+  }
+
+  __s128i(int i0, int i1, int i2, int i3) : b(0), type(SignedInt) {
+    xird[0] = i0;
+    xird[1] = i1;
+    xird[2] = i2;
+    xird[3] = i3;
+  }
+
+  __s128i(unsigned int i0, unsigned int i1,
+          unsigned int i2, unsigned int i3)
+  : b(0), type(UnsignedInt) {
+    xui[0] = i0;
+    xui[1] = i1;
+    xui[2] = i2;
+    xui[3] = i3;
+  }
+
+  __s128i(long l0, long l1)
+  : b(0), type(SignedLong) {
+    xlrd[0] = l0;
+    xlrd[1] = l1;
+  }
+
+  __s128i(unsigned long l0, unsigned long l1)
+  : b(0), type(UnsignedLong) {
+    xul[0] = l0;
+    xul[1] = l1;
+  }
+
+  __s128i(const vis4_t& v) : xird(v), type(SignedInt) { }
+
+  __s128i(const vid2_t& v) : xlrd(v), type(SignedLong) { }
+
+  __s128i(const long double& v) : b(v), type(UnsignedInt) { }
+
+  __s128i(const __s128i& rhs) : b(rhs.b), type(rhs.type) { }
+
+  inline __s128i& operator=(const __s128i& rhs) {
+    if (this != &rhs) {
+      b = rhs.b;
+      type = rhs.type;
+    }
+
+    return *this;
+  }
+
+  inline __s128i& operator=(const long double& ld) {
+    b = ld;
+    type = UnsignedInt;
+    return *this;
+  }
+
+  inline __s128i& operator=(int i) {
+    xird[0] = i;
+    xird[1] = i;
+    xird[2] = i;
+    xird[3] = i;
+    type = SignedInt;
+    return *this;
+  }
+
+  inline __s128i& operator=(unsigned int i) {
+    xird[0] = i;
+    xird[1] = i;
+    xird[2] = i;
+    xird[3] = i;
+    type = UnsignedInt;
+    return *this;
+  }
+
+  inline __s128i& operator=(const long& l) {
+    xlrd[0] = l;
+    xlrd[1] = l;
+    type = SignedLong;
+    return *this;
+  }
+
+  inline __s128i& operator=(const unsigned long& l) {
+    xul[0] = l;
+    xul[1] = l;
+    type = SignedLong;
+    return *this;
+  }
+
+  inline __s128i& operator=(const vis4_t& rhs) {
+    xird = rhs;
+    return *this;
+  }
+
+  inline __s128i& operator=(const vid2_t& rhs) {
+    xlrd = rhs;
+    return *this;
+  }
+
+  inline operator bool() const {
+    return xird[0] == 0U && xird[1] == 0U && xird[2] == 0U && xird[3] == 0U;
+  }
+
+  inline operator long double() const {
+    return b;
+  }
+
+  inline operator vis4_t() const {
+    return xird;
+  }
+
+  inline operator vid2_t() const {
+    return xlrd;
+  }
+
+  inline operator int32x4_t() const {
+    return nird;
+  }
+
+  inline operator int64x2_t() const {
+    return nlrd;
+  }
```

Comment:
This won't build with gcc8. I had do wrap it like this to make the build succeed:
```
#if __GNUC__ < 8
  inline operator int32x4_t() const {
    return nird;
  }

  inline operator int64x2_t() const {
    return nlrd;
  }
#endif
```